cplugin/source.go

```diff
@@ -32,30 +34,49 @@ type SourcePlugin interface {
 	LifecycleOnDeleted(context.Context, SourceLifecycleOnDeletedRequest) (SourceLifecycleOnDeletedResponse, error)
 }
 
+// SourceRunStream is the bidirectional stream interface for SourcePlugin.Run.
+// It combines the client and server interfaces into a single interface.
+type SourceRunStream interface {
+	// Client is only allowed to be used by the host (Conduit).
+	Client() SourceRunStreamClient
+	// Server is only allowed to be used by the plugin (connector).
+	Server() SourceRunStreamServer
+}
+
+// SourceRunStreamClient is the client-side interface for a bidirectional stream
+// of SourceRunRequest and SourceRunResponse messages.
+type SourceRunStreamClient interface {
+	Send(SourceRunRequest) error
+	Recv() (SourceRunResponse, error)
+}
+
+// SourceRunStreamServer is the server-side interface for a bidirectional stream
+// of SourceRunRequest and SourceRunResponse messages.
+type SourceRunStreamServer interface {
+	Send(SourceRunResponse) error
+	Recv() (SourceRunRequest, error)
+}
```

Comment:
The stream handling was changed a bit. The reason is that we want to retain the same interface for the plugin on the client side and server side, so we can skip the whole transformation layer we had before.